### PR TITLE
Move TCK from smallrye

### PR DESCRIPTION
**Pull request by brunobat**

### CODE REVIEW

File: tracing/tck/src/main/resources/META-INF/NOTICE [hunk 1]
```diff
@@ -0,0 +1,13 @@
+=========================================================================
+==  NOTICE file corresponding to section 4(d) of the Apache License,   ==
+==  Version 2.0, in this case for Microprofile JWT Auth                  ==
```
Comment (Emily-Jiang) [inline tracing/tck/src/main/resources/META-INF/NOTICE:3]:
```suggestion
==  Version 2.0, in this case for MicroProfile Telemetry                  ==
```

File: tracing/tck/pom.xml [hunk 2]
```diff
@@ -23,14 +24,113 @@
...
+                        <configuration>
+                            <sourceFile>../README.adoc</sourceFile>
+                            <destinationFile>target/classes/README.adoc</destinationFile>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
 </project>
```
Comment (brunobat) [inline tracing/tck/pom.xml:136]: add line

File: pom.xml [hunk 3]
```diff
@@ -25,13 +26,13 @@
...
     <artifactId>microprofile-telemetry-parent</artifactId>
     <version>1.0-SNAPSHOT</version>
-
     <packaging>pom</packaging>
     <name>MicroProfile Telemetry</name>
     <url>https://microprofile.io/project/eclipse/microprofile-telemetry</url>
 
     <properties>
         <inceptionYear>2022</inceptionYear>
+        <opentelemetry-api.version>1.14.0</opentelemetry-api.version>
```
Comment (Emily-Jiang) [inline pom.xml:35]:
```suggestion
        <opentelemetry-api.version>1.15.0</opentelemetry-api.version>
```

File: tracing/api/pom.xml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>jakarta.enterprise</groupId>
+            <artifactId>jakarta.enterprise.cdi-api</artifactId>
+            <scope>provided</scope>
+        </dependency>
+    </dependencies>
+
+</project>
```
Comment (Emily-Jiang) [inline tracing/api/pom.xml:42]: This needs to be removed as per our discussion.

File: tracing/api/src/main/java/org/eclipse/microprofile/telemetry/tracing/api/TelemetryTracingConfig.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+@ApplicationScoped
+public class TelemetryTracingConfig {
+
+    @Inject
+    @ConfigProperty(name = "mp.telemetry.tracing.name", defaultValue = "org.eclipse.microprofile.telemetry.tracing.api")
+    public String instrumentationName;
+
+    @Inject
+    @ConfigProperty(name = "mp.telemetry.tracing.version", defaultValue = "not set")
+    public String instrumentationVersion;
```
Comment (Emily-Jiang) [inline tracing/api/src/main/java/org/eclipse/microprofile/telemetry/tracing/api/TelemetryTracingConfig.java:39]: This file is to be deleted as per our discussion.

File: tracing/pom.xml [hunk 6]
```diff
@@ -31,4 +32,59 @@
         <module>tck</module>
     </modules>
 
+    <properties>
+        <!--        <java.version>11</java.version>-->
+        <!--        <maven.compiler.target>${java.version}</maven.compiler.target>-->
+        <!--        <maven.compiler.source>${java.version}</maven.compiler.source>-->
+
+        <version.opentelemetry>1.12.1</version.opentelemetry>
```
Comment (Emily-Jiang) [inline tracing/pom.xml:40]: Where is this used?

File: tracing/spec/src/main/asciidoc/microprofile-telemetry-tracing-spec.asciidoc [hunk 7]
```diff
@@ -248,6 +248,10 @@ attribute is optional. Implementations can also add their own attributes.
 Other MicroProfile specifications can add their own attributes under their own attribute name following the
 convention `mp.[specification short name].[attribute name]`.
 
+Implementation libraries must set the library name and version using the following properties:
+`mp.telemetry.tracing.name`
```
Comment (Emily-Jiang) [inline tracing/spec/src/main/asciidoc/microprofile-telemetry-tracing-spec.asciidoc:252]: I was told these properties are not required in the latest release of Open Telemetry. We might not need it any more.

Comment (Emily-Jiang) [reply tracing/spec/src/main/asciidoc/microprofile-telemetry-tracing-spec.asciidoc:252]: or it can be changed from must to may.

**Comment by Emily-Jiang** [comment]
@yasmin-aumeeruddy pointed out 1.15 has not got instrumentation jar released, so we are better off to stay with 1.14.

**Comment by fmhwong** [comment]
I built this branch and got the following checkstyle errors.

```
[INFO] There are 7 errors reported by Checkstyle 8.29 with /Users/fmhwong/gitRepos/microprofile-telemetry/tracing/tck/target/checkstyle-rules.xml ruleset.
[ERROR] src/main/Java/org/eclipse/microprofile/telemetry/tracing/tck/exporter/InMemorySpanExporter.java:[25,15] (imports) UnusedImports: Unused import - org.awaitility.Awaitility.await.
[ERROR] src/main/Java/org/eclipse/microprofile/telemetry/tracing/tck/exporter/InMemorySpanExporter.java:[26,15] (imports) UnusedImports: Unused import - org.junit.jupiter.api.Assertions.assertEquals.
[ERROR] src/main/Java/org/eclipse/microprofile/telemetry/tracing/tck/TestConfigSource.java:[30,38] (naming) ConstantName: Name 'configuration' must match pattern '^(([A-Z][A-Z0-9]*(_[A-Z0-9]+)*))$'.
[ERROR] src/main/Java/org/eclipse/microprofile/telemetry/tracing/tck/rest/RestSpanTest.java:[26] (imports) AvoidStarImport: Using the '.*' form of import should be avoided - io.opentelemetry.semconv.trace.attributes.SemanticAttributes.*.
[ERROR] src/main/Java/org/eclipse/microprofile/telemetry/tracing/tck/rest/RestSpanTest.java:[49] (imports) AvoidStarImport: Using the '.*' form of import should be avoided - jakarta.ws.rs.*.
[ERROR] src/main/Java/org/eclipse/microprofile/telemetry/tracing/tck/rest/RestClientSpanTest.java:[36,15] (imports) UnusedImports: Unused import - org.junit.jupiter.api.Assertions.assertEquals.
[ERROR] src/main/Java/org/eclipse/microprofile/telemetry/tracing/tck/rest/RestClientSpanTest.java:[63] (imports) AvoidStarImport: Using the '.*' form of import should be avoided - jakarta.ws.rs.*.
[WARNING] checkstyle:check violations detected but failOnViolation set to false
```

File: tracing/pom.xml (hunk 6 above)
Comment (brunobat) [reply tracing/pom.xml:40]: Nowhere. Will delete

**Comment by JanWesterkamp-iJUG** [comment]
@brunobat I reviewed this PR including your last changes (looks good), but as dicussed in the last meeting, I am not a official contributor to this repo yet. So I am unable to merge the PR at the moment.
@Emily-Jiang, @brunobat can somebody give me the necessary rights?
Then I can merge afterwards.

**Review by Emily-Jiang — APPROVED** [review]
LGTM! Thanks you @brunobat

**Comment by Emily-Jiang** [comment]
> @brunobat I reviewed this PR including your last changes (looks good), but as dicussed in the last meeting, I am not a official contributor to this repo yet. So I am unable to merge the PR at the moment. @Emily-Jiang, @brunobat can somebody give me the necessary rights? Then I can merge afterwards.

@JanWesterkamp-iJUG [this](https://waynebeaton.wordpress.com/2020/04/07/becoming-an-eclipse-committer/) blog explains how to become an Eclipse committer.

**Pull request merged**